go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -381,18 +386,21 @@ func allocatePodAnnotationWithRollback(
 
 	if hasIPAM {
 		if len(tentative.IPs) > 0 {
-			if err = ipAllocator.AllocateIPs(tentative.IPs); err != nil && !ip.IsErrAllocated(err) {
-				err = fmt.Errorf("failed to ensure requested or annotated IPs %v for %s: %w",
-					util.StringSlice(tentative.IPs), podDesc, err)
-				if !reallocateOnNonStaticIPRequest {
-					return
+			err = ipAllocator.AllocateIPs(tentative.IPs)
+			if err != nil {
+				if !ip.IsErrAllocated(err) || !shouldIgnoreDuplicateIPError(podLister, pod, nadNetworkPersisted) {
+					err = fmt.Errorf("failed to ensure requested or annotated IPs %v for %s: %w",
+						util.StringSlice(tentative.IPs), podDesc, err)
+					if !reallocateOnNonStaticIPRequest {
+						return
+					}
+					klog.Warning(err.Error())
+					needsIPOrMAC = true
+					tentative.IPs = nil
 				}
-				klog.Warning(err.Error())
-				needsIPOrMAC = true
-				tentative.IPs = nil
 			}
 
-			if err == nil && !hasIPAMClaim { // if we have persistentIPs, we should *not* release them on rollback
+			if err == nil && (!hasIPAMClaim || !nadNetworkPersisted) { // if we have persistentIPs, we should *not* release them on rollback, unless the annotation isn't persisted yet
```

Comment:
Don't we have to check here exactly the same we check at the allocat error skipping mechanism, for example we should not rollback also if we are on live migration.

Reply:
hmmm I think it makes sense, during migration there is no real IP allocation, the IP only moves to another pod. So better not release the IP in this case. 
The concern is of the following scenario:
1. target pod created, in first reconcile err is ignored
2. even after network annotation persisted, we still need to ignore until the migration is complete